plugins/system76-launch/fu-system76-launch-device.c

```diff
@@ -48,18 +61,23 @@ fu_system76_launch_device_command(FuDevice *device, guint8 *data, gsize len, GEr
 		return FALSE;
 	}
 
-	/* receive response */
-	if (!g_usb_device_interrupt_transfer(usb_device,
-					     ep_in,
-					     data,
-					     len,
-					     &actual_len,
-					     SYSTEM76_LAUNCH_TIMEOUT,
-					     NULL,
-					     error)) {
-		g_prefix_error(error, "failed to read response: ");
-		return FALSE;
-	}
+	do {
+		/* receive response */
+		if (!g_usb_device_interrupt_transfer(usb_device,
+						     ep_in,
+						     data,
+						     len,
+						     &actual_len,
+						     SYSTEM76_LAUNCH_TIMEOUT,
+						     NULL,
+						     error)) {
+			g_prefix_error(error, "failed to read response: ");
+			return FALSE;
+		}
+
+		attempt++;
+	} while (actual_len == 0 && attempt < 5);
```

Comment:
can you use fu_device_retry() here?

Reply:
I will try

Reply:
I think fu_device_retry() would be easier to read fwiw. Do you want me to convert it?

Reply:
That would be helpful, thanks.

Reply:
does 4038a90bbe07d2fce898cf3d491f3afd45dcdb18 work? If so, can you squash that into the last commit and force push again pls.

Reply:
Thanks, I am testing now.

Reply:
It works, I squashed it as requested.